portfolio/src/main/webapp/script.js

```diff
@@ -109,22 +178,164 @@ function retrieveId(string, source){
 }
 
 /**
-  * Runs the search by getting the source HTML, getting the appropriate tag, and scrolling towards that id
-  */
+ * Get's all id's from a given tag array
+ */
+function getIds(array, source){
+    var retArr = [];
+    for (var i = 0; i < array.length; i++){
+        var retriever = retrieveId(array[i], source);
+        if (retriever === "nothing"){
+            continue;
+        } else {
+            retArr.push(retriever);
+        }
```

Comment:
this can be condensed to something like

if (result is found) { retArr.push(result) }